src/cascadia/TerminalSettingsEditor/ColorSchemes.cpp

```diff
@@ -116,11 +127,9 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
     // - <none>
     void ColorSchemes::_UpdateColorTable(const Model::ColorScheme& colorScheme)
     {
-        _CurrentColorTable.Clear();
         for (uint8_t i = 0; i < TableColorNames.size(); ++i)
         {
-            auto entry = winrt::make<ColorTableEntry>(i, colorScheme.Table()[i]);
-            _CurrentColorTable.Append(entry);
+            _CurrentColorTable.GetAt(i).Color(colorScheme.Table()[i]);
```

Comment:
_interesting_. So if we did `SetAt()` here, we would run into the same problem? Then, we don't actually want `_CurrentColorTable` to be an `IObservableVector`, right? Could we just make that a regular `IVector`?

Reply:
probably could!

Reply:
Yup, doing `SetAt()` also gives the same kind of wonky behavior 😢. I think you might be right lemme try doing `IVector`.

Reply:
it... crashes

Reply:
oh it might be because in order to use `IVector` in data binding it needs to be `IVector<IInspectable>`